src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/UriContent.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Text.Json.Serialization;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Represents a URL, typically to hosted content such as an image, audio, or video.
+/// </summary>
+/// <remarks>
+/// This class is intended for use with HTTP or HTTPS URIs that reference hosted content.
+/// For data URIs, use <see cref="DataContent"/> instead.
+/// </remarks>
+[DebuggerDisplay("{DebuggerDisplay,nq}")]
+public class UriContent : AIContent
+{
+    /// <summary>The URI represented.</summary>
+    private Uri _uri;
+
+    /// <summary>The MIME type of the data at the referenced URI.</summary>
+    private string _mediaType;
+
+    /// <summary>Initializes a new instance of the <see cref="UriContent"/> class.</summary>
+    /// <param name="uri">The URI to the represented content.</param>
+    /// <param name="mediaType">The media type (also known as MIME type) represented by the content.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="uri"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentNullException"><paramref name="mediaType"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException"><paramref name="mediaType"/> is an invalid media type.</exception>
+    /// <exception cref="UriFormat"><paramref name="uri"/> is an invalid URL.</exception>
+    /// <remarks>
+    /// A media type must be specified, so that consumers know what to do with the content.
+    /// If an exact media type is not known, but the category (e.g. image) is known, a wildcard
+    /// may be used (e.g. "image/*").
+    /// </remarks>
+    public UriContent(string uri, string mediaType)
+        : this(new Uri(Throw.IfNull(uri)), mediaType)
```

Comment:
Should we consider throwing if `uri` is a data URI, forcing people to use `DataContent` for those?

I'm guessing you didn't because a data URI *is* a URI and hence there's no strict reason why we should block it here.

Reply:
The only payoff to being more constrained is if we think `IChatClient` implementations are likely not to understand the case where `UriContent` is a data URI and then for something to go wrong. For example an `IChatClient` implementation might be forced by its own upstream APIs to distinguish between "remote URIs" and "data URIs" and might incorrectly use our `UriContent`/`DataContent` distinction to model that.

I think we don't really gain much by imposing an extra restriction here. If the `IChatClient` implementation does something wrong, that's their bug and they should fix it.

So overall no, I'd vote with keeping it as-is.